addons/mail_bot/models/res_users.py

```diff
@@ -16,6 +16,8 @@ class Users(models.Model):
             ('disabled', 'Disabled'),
         ], string="OdooBot Status", readonly=True, required=True, default="not_initialized")  # keep track of the state: correspond to the code of the last message sent
 
+    odoobot_failed = fields.Boolean(readonly=True, default=False)
```

Comment:
I don't really understand, what is the purpose of this field ? Same as for mail.message, I am really not a fan of adding fields just for corner case of a small corner application ...

Reply:
It serves as a way of keeping track of whether the user has failed the current stage of the odoobot tour, as it was in the specs that if the user fails 2 or more times the same step, a different reply message should be used.

Reply:
Ok :) indeed I did not find any better solution